File: part2/4_simple_linear_regression/simple_linear_regression.R
# Simple Linear regression
dataset = read.csv('Salary_Data.csv')

library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

regressor = lm(formula = Salary ~ YearsExperience,
               data = training_set)
# to get more info about the regressor, in console type:
# summary(regressor)

#Coefficients:
#  Estimate Std. Error t value Pr(>|t|)    
#(Intercept)      25792.2     2273.1   11.35 5.51e-12 ***
#  YearsExperience   9450.0      378.8   24.95  < 2e-16 ***

y_pred = predict(regressor, newdata=test_set)

# Visualizing the training set results
# install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point(data = training_set, aes(x=YearsExperience,y=Salary), color='red') +
  geom_line(data = training_set, aes(x=YearsExperience,y=predict(regressor, newdata = training_set)), color='blue') +
  ggtitle('Salary versus experience') +
  xlab('Years of experience') +
  ylab('Salary')

# Visualizing the test set results
# install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point(data = test_set, aes(x=YearsExperience,y=Salary), color='red') +
  geom_line(data = training_set, aes(x=YearsExperience,y=predict(regressor, newdata = training_set)), color='blue') +
  ggtitle('Salary versus experience') +
  xlab('Years of experience') +
  ylab('Salary')

File: part2/4_simple_linear_regression/simple_linear_regression.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Sat Feb 17 10:34:19 2018

@author: chen
"""

# Simple

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:,:-1].values
y = dataset.iloc[:,1].values

# splitting the data into train and test
from sklearn.cross_validation import train￼_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=1/3, random_state=0)

# Fitting simple linear regression to the training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

# Predicting the test set results
y_pred = regressor.predict(X_test)

# Visualizing the training set results
plt.scatter(X_train, y_train, color='red')
plt.plot(X_train, regressor.predict(X_train), color='blue')
plt.title('Salary vs Experience - training set')
plt.xlabel('Years of experience')
plt.ylabel('Salary')
plt.show()

# Visualizing the test set result
plt.scatter(X_test, y_test, color='red')
plt.plot(X_train, regressor.predict(X_train), color='blue')
plt.title('Salary vs Experience - test set')
plt.xlabel('Years of experience')
plt.ylabel('Salary')
plt.show()